### ISSUE TITLE
error during exporting model AssertionError: Each dimension size given in out_shape must be greater than 0.

### ISSUE THREAD

**Issue by DXVVV** [PaddlePaddle/PaddleSeg#1498]
我在导出模型，然后用下面的bash命令
```
python export.py \
	--config configs/ginet/myconfig.yml \
	--model_path gi_output/best_model/model.pdparams
```
但是报了如下的错，我是在自制数据集上训练的，训练和评估过程一切正常，但是就是导出有错，换了几个模型，2.3新增的模型就是不行，老的ok。
---
platform: Linux-4.15.0-142-generic-x86_64-with-debian-stretch-sid
Python: 3.7.0 (default, Jun 28 2018, 13:15:42) [GCC 7.2.0]
Paddle compiled with cuda: True
NVCC: Cuda compilation tools, release 10.2, V10.2.89
cudnn: 7.6
GPUs used: 1
CUDA_VISIBLE_DEVICES: 0,1
GPU: ['GPU 0: Tesla K80', 'GPU 1: Tesla K80']
GCC: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
PaddlePaddle: 2.1.0
OpenCV: 4.4.0
W1104 22:47:22.849423 31894 device_context.cc:404] Please NOTE: device: 0, GPU Compute Capability: 3.7, Driver API Version: 10.2, Runtime API Version: 10.2
W1104 22:47:22.849475 31894 device_context.cc:422] device: 0, cuDNN Version: 7.6.
2021-11-04 22:47:24 [INFO]	

---
error log:
```
/home/muli/anaconda3/lib/python3.7/site-packages/socks.py:58: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  from collections import Callable
W1104 22:15:02.934046  9295 device_context.cc:404] Please NOTE: device: 0, GPU Compute Capability: 3.7, Driver API Version: 10.2, Runtime API Version: 10.2
W1104 22:15:02.938066  9295 device_context.cc:422] device: 0, cuDNN Version: 7.6.
2021-11-04 22:15:05 [INFO]	Loading pretrained model from https://bj.bcebos.com/paddleseg/dygraph/resnet50_vd_ssld_v2.tar.gz
2021-11-04 22:15:06 [INFO]	There are 275/275 variables loaded into ResNet_vd.
2021-11-04 22:15:08 [INFO]	Loaded trained params of model successfully.
/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/layers/utils.py:77: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  return (isinstance(seq, collections.Sequence) and
Traceback (most recent call last):
  File "export.py", line 143, in <module>
    main(args)
  File "export.py", line 122, in main
    paddle.jit.save(new_net, save_path)
  File "<decorator-gen-67>", line 2, in save
  File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/wrapped_decorator.py", line 25, in __impl__
    return wrapped_func(*args, **kwargs)
  File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/base.py", line 40, in __impl__
    return func(*args, **kwargs)
  File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/jit.py", line 719, in save
    inner_input_spec)
  File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/dygraph_to_static/program_translator.py", line 488, in concrete_program_specify_input_spec
    *desired_input_spec)
  File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/dygraph_to_static/program_translator.py", line 402, in get_concrete_program
    concrete_program, partial_program_layer = self._program_cache[cache_key]
  File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/dygraph_to_static/program_translator.py", line 711, in __getitem__
    self._caches[item] = self._build_once(item)
  File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/dygraph_to_static/program_translator.py", line 702, in _build_once
    class_instance=cache_key.class_instance)
  File "<decorator-gen-65>", line 2, in from_func_spec
  File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/wrapped_decorator.py", line 25, in __impl__
    return wrapped_func(*args, **kwargs)
  File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/base.py", line 40, in __impl__
    return func(*args, **kwargs)
  File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/dygraph_to_static/program_translator.py", line 658, in from_func_spec
    error_data.raise_new_exception()
  File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/dygraph_to_static/error.py", line 192, in raise_new_exception
    six.exec_("raise new_exception from None")
  File "<string>", line 1, in <module>
AssertionError: In transformed code:

    File "export.py", line 74, in forward (* user code *)
	outs = self.net(x)
    File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/layers.py", line 898, in __call__
	outputs = self.forward(*inputs, **kwargs)
    File "/home/muli/xiang/paddle/PaddleSeg/paddleseg/models/ginet.py", line 83, in forward (* user code *)
	_, _, c3, c4 = self.base_forward(x)
    File "/tmp/tmptslb47y4.py", line 39, in base_forward
	__return_value_0,))
    File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/dygraph_to_static/convert_operators.py", line 210, in convert_ifelse
	return _run_py_ifelse(pred, true_fn, false_fn, true_args, false_args)
    File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/dygraph_to_static/convert_operators.py", line 235, in _run_py_ifelse
	return true_fn(*true_args) if pred else false_fn(*false_args)
    File "/home/muli/xiang/paddle/PaddleSeg/paddleseg/models/ginet.py", line 77, in base_forward (* user code *)
	return self.jpu(c1, c2, c3, c4)
    File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/layers.py", line 898, in __call__
	outputs = self.forward(*inputs, **kwargs)
    File "/home/muli/xiang/paddle/PaddleSeg/paddleseg/models/layers/layer_libs.py", line 259, in forward (* user code *)
	feats[-2], size, mode='bilinear', align_corners=True)
    File "/home/muli/anaconda3/lib/python3.7/site-packages/paddle/nn/functional/common.py", line 365, in interpolate
	"Each dimension size given in out_shape must be greater than 0."
    _base_: '../_base_/pascal_voc12aug.yml'

model:
  type: GINet
  backbone:
    type: ResNet50_vd
    output_stride: 8
    pretrained: https://bj.bcebos.com/paddleseg/dygraph/resnet50_vd_ssld_v2.tar.gz
  backbone_indices: [0, 1, 2, 3]
  enable_auxiliary_loss: True
  jpu: True
  align_corners: True
  pretrained: null

loss:
  types:
    - type: CrossEntropyLoss
  coef: [1, 0.4]


optimizer:
  type: sgd
  momentum: 0.9
  weight_decay: 0.0005

train_dataset: #训练数据设置
  type: Dataset #选择数据集格式
  dataset_root: data/custom_dataset #选择数据集路径
  train_path: data/custom_dataset/train.txt
  num_classes: 6 #指定目标的类别个数（背景也算为一类）
  transforms: #数据预处理/增强的方式
    - type: Resize #送入网络之前需要进行resize
      target_size: [512, 512] #将原图resize成512*512在送入网络
    - type: RandomHorizontalFlip #采用水平反转的方式进行数据增强
    - type: Normalize #图像进行归一化
  mode: train

val_dataset: #验证数据设置
  type: Dataset #选择数据集格式
  dataset_root: data/custom_dataset #选择数据集路径
  val_path: data/custom_dataset/val.txt
  num_classes: 6 #指定目标的类别个数（背景也算为一类）
  transforms: #数据预处理/增强的方式
    - type: Resize  #将原图resize成512*512在送入网络
      target_size: [512, 512]  #将原图resize成512*512在送入网络
    - type: Normalize #图像进行归一化
  mode: val.

```
---
config:
```
_base_: '../_base_/pascal_voc12aug.yml'

model:
  type: GINet
  backbone:
    type: ResNet50_vd
    output_stride: 8
    pretrained: https://bj.bcebos.com/paddleseg/dygraph/resnet50_vd_ssld_v2.tar.gz
  backbone_indices: [0, 1, 2, 3]
  enable_auxiliary_loss: True
  jpu: True
  align_corners: True
  pretrained: null

loss:
  types:
    - type: CrossEntropyLoss
  coef: [1, 0.4]


optimizer:
  type: sgd
  momentum: 0.9
  weight_decay: 0.0005

train_dataset: #训练数据设置
  type: Dataset #选择数据集格式
  dataset_root: data/custom_dataset #选择数据集路径
  train_path: data/custom_dataset/train.txt
  num_classes: 6 #指定目标的类别个数（背景也算为一类）
  transforms: #数据预处理/增强的方式
    - type: Resize #送入网络之前需要进行resize
      target_size: [512, 512] #将原图resize成512*512在送入网络
    - type: RandomHorizontalFlip #采用水平反转的方式进行数据增强
    - type: Normalize #图像进行归一化
  mode: train

val_dataset: #验证数据设置
  type: Dataset #选择数据集格式
  dataset_root: data/custom_dataset #选择数据集路径
  val_path: data/custom_dataset/val.txt
  num_classes: 6 #指定目标的类别个数（背景也算为一类）
  transforms: #数据预处理/增强的方式
    - type: Resize  #将原图resize成512*512在送入网络
      target_size: [512, 512]  #将原图resize成512*512在送入网络
    - type: Normalize #图像进行归一化
  mode: val
```

**Comment by shiyutang** [comment]
你好，这个是由于动转静过程中使用reshape操作导致形状纬度-1污染tensor报错，目前已经修改，请参考[PR](https://github.com/PaddlePaddle/PaddleSeg/pull/1516#issue-776158135)

**Comment by DXVVV** [comment]
> 染

你好，谢谢你们及时的反馈，值得一提的是，我尝试了pointrend，也是有同样的问题，我看你的修改只是ginet的文件，可以麻烦查看一下pointrend是否有同样的问题呢

**Comment by shiyutang** [comment]
Sure, we will look at it and get back to you ASAP~

**Comment by shiyutang** [comment]
Please look at pointrend at the develop branch, we have fix the export problem of pointrend in PR #1449  
I just export & infer pointrend with code in develop, and successfully get the inference result.
![image](https://user-images.githubusercontent.com/34859558/142348923-42205dc1-d251-4857-afba-a00d01ed90a1.png)

![image](https://user-images.githubusercontent.com/34859558/142348870-01032698-0e67-4fb7-b0d9-07b07213b443.png)
![image](https://user-images.githubusercontent.com/34859558/142348996-e32ad5ee-1993-4a1d-93d2-004868d35e5c.png)

**Comment by DXVVV** [comment]
没问题了，谢谢
